katana/native/file_manager/file_manager_utils/ftp_utils.py

```diff
@@ -0,0 +1,153 @@
+    :param destdir: Location from the root directory in the Target Machine where the files should be FTPed into
+    :param files_path: Full path of the top level files/directories selected
+    :param files_name: Names of the top level files/directories selected
+    :param all_files_path: List of All the files/directories selected (due to down cascading)
+    :return: Error message or list of files FTPed with status.
+    """
+    result = []
+    os.chdir(getpath(request))
+    try:
+        ftp = FTP('')
+        ftp.connect(host, port)
+        ftp.login(username, passwd)
+        try:
+            if(change_destination(ftp, destdir)):
+                for path, name in zip(files_path, files_name):
+                    if os.path.isfile(path):
+                        result.append(transfer_file(ftp, path, name))
+                    elif os.path.isdir(path):
+                        result.append(name + ': Is a directory')
+                        result.append(isDir(ftp, path, name, files_path, files_name, all_files_path))
+                    else:
+                        result.append(name + ': Is not a file or directory')
+            else:
+                result.append(destdir + " :Could Not Change to Destination Directory Successfully")
+                return result
+        except ftplib.all_errors as e:
+            result.append(str(e))
+        ftp.quit()
+    except ftplib.all_errors as e:
+        result.append(str(e))
+    return result
+
+
+def change_destination(ftp, destdir):
+    """
+    If the destiantion directory is different than the root in the target machine, then this function traverses into
+    the destination directory in the target machine.
+
+    :param ftp: FTP object
+    :param destdir: Destination directory from the root directory where the files are to be FTPed.
+    :return: Boolean.
+    """
+    try:
+        if destdir != '':
+            ftp.cwd(destdir)
+    except:
+        print('\nCould not change to destination directory')
+        return False
+    return True
+
+
+def transfer_file(ftp, path, name):
+    """
+    This function FTPs the file. Some files have to be sent as binary format and some as ascii format.
+
+    :param ftp: FTP object
+    :param path: Full path to file name
+    :param name: File name
+    :return: Status of the file transfer
+    """
+    fp = open(path, 'rb')
+    try:
+        ext = os.path.splitext(name)[1]
+        # check if the extension needs to be sent in binary mode. Default is ASCII mode.
+        if ext == '.docx' or ext == '.doc':
+            ftp.storbinary('STOR ' + name, fp)
+        else:
+            ftp.storlines('STOR ' + name, fp)
+        # storlines prints new line character at the end of each line in text file
+        # but it does not transfer word files.
+    except:
+        return name + ': Could not transfer the file.'
+    fp.close()
+    return name + ' :Success'
+
+
+def isDir(ftp, path, name, files_path, files_name, all_files_path):
```

Comment:
Please change function name to ```is_dir``` so that it follows pylint conventions.

Reply:
Done